scrapers_next/ca/committees.py

```diff
@@ -0,0 +1,343 @@
+from spatula import HtmlListPage, URL, XPath, CSS, HtmlPage, SelectorError
+from openstates.models import ScrapeCommittee
+import re
+
+
+class ChooseType(HtmlPage):
+    def process_page(self):
+
+        # this link was being considered type one, but should be type four
+        if self.source.url == "https://ajed.assembly.ca.gov":
```

Comment:
OK I've been thinking about this and I think I have an alternate solution that I'd recommend that'll be some small code refactoring but fortunately no changes to the actual functions below

instead of TypeOne through TypeFour being their own classes, could we go with

def process_page_type1(root, committee, source, ...):
      # existing code from TypeOne.process_item with `self` usage replaced with parameters above
      return com 

(and so on)

and then instead of returning TypeOne/TypeTwo/TypeThree/etc. you could return the result of these functions

if this doesn't make sense, we can talk it through, as it is a bit unorthodox, but will reduce requests by 50%

Reply:
I'm not sure I understand this. Could we meet at some point and talk this through?

Reply:
I think I was able to figure this out. Let me know what you think. Only needed to pass source for process_page_type4()